### DISCUSSION TITLE
Shopify event tickets delivered before the actual payment happens

### DISCUSSION THREAD

**Discussion by 77elements** [btcpayserver/btcpayserver#6064] [Plugins & integrations]
We're selling event tickets at our Shopify store

The customer gets the ticket without paying for it. Here are the steps to reproduce it:

1. When customer clicks "Buy Tickets" button 
2. Go to checkout, choose "Bitcoin with BTCPay Server" as payment option. It will take you to the Review Order page. 
![btcpay-screenshot01-1](https://github.com/btcpayserver/btcpayserver/assets/7952176/45cda18e-8f50-41cb-b627-42db598b3849)

At this step, the customer already gets a order confirmation email. 
![btcpay-screenshot02-2](https://github.com/btcpayserver/btcpayserver/assets/7952176/1279e20e-c436-4434-bd99-620982f91a4a)

3. When I click on that big blue button in the order condirmation email, I'm directed to the Review Order page again
4. There, click on "Complete registration" ...
5. It takes me to a page asking for the attendee's name. 
![btcpay-screenshot03-1](https://github.com/btcpayserver/btcpayserver/assets/7952176/dcb5fcfe-3029-4af8-a92e-6ee2f38d2ad7)

6. It takes the customer back to the Review Order page, but this time he can download the ticket.  
![btcpay-screenshot04-2](https://github.com/btcpayserver/btcpayserver/assets/7952176/55167614-1814-4ddf-9021-5543865de3fe)

I'm running BTCPay Server v1.13.3 on a voltage.cloud server.

We already talked to the event plugin developer, he said his plugin is not involved in the checkout process and referred su to Shopify support. Shopify support couldn't help us either.

**Comment by Kukks** [comment]
Most likely, the ticket plugin is not checking if the order has been paid. Btcpay's integration has to use some workarounds due to shopify's unfriendly attitude towards payment processors that are self hosted (and they therefore cannot charge us or you or extract payment info).

In BTCPay the same order follows the same flow as if you were accepting a cash payment in shopify. The order will move along to the next step but it will still be marked as unpaid in the administrative side.

Whatever is issuing tickets needs to also be checking if the order is paid (which we only do after we confirm funds have been received and settled)
